From d2c2545b7bd3e2f325eeb66ded8d9d243d933545 Mon Sep 17 00:00:00 2001
From: Zhuang Jin Can <jin.can.zhuang@intel.com>
Date: Tue, 5 May 2015 15:56:21 +0800
Subject: [PATCH] workaround: usb: quirk: void reset-resume for 7262 SSIC modem

Most of the time, reset-resume fails to rescue the modem.
Even it succeed to rescue the link, sometimes modem fails to accept
the new address.

Sometimes, after reset-resume the port is stuck in reset state.

Moreover, it's not clear how reset-resume works for SSIC PHY. So
keep it simple and don't try to reset-resume.

Change-Id: Ie228ebabca1a0f1549720d4de18b4ea108f9d75f
Signed-off-by: Zhuang Jin Can <jin.can.zhuang@intel.com>
---
 drivers/usb/core/quirks.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/usb/core/quirks.c b/drivers/usb/core/quirks.c
index b195fdb..355e753 100644
--- a/drivers/usb/core/quirks.c
+++ b/drivers/usb/core/quirks.c
@@ -165,6 +165,9 @@ static const struct usb_device_id usb_quirk_list[] = {
 	/* INTEL VALUE SSD */
 	{ USB_DEVICE(0x8086, 0xf1a5), .driver_info = USB_QUIRK_RESET_RESUME },
 
+	/* INTEL SSIC modem */
+	{ USB_DEVICE(0x1519, 0x0452), .driver_info = USB_QUIRK_RESET },
+
 	{ }  /* terminating entry must be last */
 };
 
-- 
1.9.1

